magma/array.py

```diff
@@ -89,6 +89,19 @@ def __getitem__(cls, index: tuple) -> 'ArrayMeta':
 
         # index[0] (N) can be None (used internally for In(Array))
         if index[0] is not None:
+            if isinstance(index[0], tuple):
```

Comment:
technically, I don't think anything would break...but should we constrain what `T` can be? Specifically if `T` is itself some array or nested type, its unclear what the semantics should be. if `T` is array, should that be sliceable in the same way too? basically are `Array[N, Array[M, T]]` and `Array[(N, M), T]` the same?

Reply:
Right now those are treated as the same, so the ndarray logic just assumes a container of type `Array[M, Array[N, ...]]`.  The meta class logic unpacks `Array[(M, N)]` into the above form and the tuple indexing logic assumes it has the corresponding structure.  So the above example will work, and while confusing, not sure if we need to constrain it

Reply:
I think that's ideal actually. It would be even more confusing if they resulted in different Thanks for clarifying.